src/DurableTask.AzureStorage/AzureStorageOrchestrationService.cs

```diff
@@ -503,18 +513,62 @@ internal async Task OnOwnershipLeaseAquiredAsync(BlobLease lease)
             this.allControlQueues[lease.PartitionId] = controlQueue;
         }
 
+        internal void DropLostControlQueues(TableLease partition)
```

Comment:
nit: we're only removing one control queue at a time, so let's rename this accordingly. You might need to update the places where this method is called too.
```suggestion
        internal void DropLostControlQueue(TableLease partition)
```